client/express-checkout/event-handlers.js

```diff
@@ -33,8 +35,6 @@ export const shippingAddressChangeHandler = async ( api, event, elements ) => {
 				amount: response.total.amount,
 			} );
 
-			lastSelectedAddress = event.address;
```

Comment:
Moved above, so that the newly selected address is saved regardless of the error.
This makes the behavior consistent between block-based and shortcode-based checkout.